### PR TITLE
Make `Lint/SharedVarInFiber` rule account for `loop { ... }`

### LINKED ISSUE #353: Using `loop do` hides `SharedVarInFiber` warning
Using ameba 1.4.0 (originally found with 0.14.3):

This code generates, properly, a `SharedVarInFiber` warning:
`a.cr`:
```
x = 1
while true
  x = x + 1
  spawn do
    puts x
  end
end
```

Yet this code (functionally identical) doesn't warn about `SharedVarInFiber`:
```
x = 1
loop do
  x = x + 1
  spawn do
    puts x
  end
end
```

Here is `ameba` inspecting the two different files:
```
$ ameba a.cr
Inspecting 1 file

F

a.cr:5:10
[W] Lint/SharedVarInFiber: Shared variable `x` is used in fiber
> puts x
       ^

a.cr:2:1 [Correctable]
[C] Style/WhileTrue: While statement using true literal as condition
> while true
  ^

Finished in 837 microseconds
1 inspected, 2 failures
$ vim b.cr
$ ameba b.cr
Inspecting 1 file

.

Finished in 767 microseconds
1 inspected, 0 failures
$
```

### PR DESCRIPTION
**Pull request by Sija**
Closes #353

### CODE REVIEW

File: src/ameba/ast/branch.cr [hunk 1]
```diff
@@ -172,6 +176,16 @@ module Ameba::AST
...
       end
+
+      def visit(node : Crystal::Call)
+        if loop?(node) && (block = node.block)
+          on_branchable_start node, block.body
+        end
+      end
+
+      def end_visit(node : Crystal::Call)
+        on_branchable_end node if loop?(node)
```
Comment (veelenga) [inline src/ameba/ast/branch.cr:187]: i think this `if` condition should match the one in visit method because it can mess up the scope

**Review by veelenga — APPROVED** [review]
LGTM

**Comment by veelenga** [comment]
Oops, auto-merge doesn't wait for the CI status :/

**Comment by Sija** [comment]
@veelenga I've made the `test (<os>, latest)` checks required, let's see if that helps.

**Pull request merged**